src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2056,12 +1989,11 @@ SubquerySqlTaskList(Job *job)
 			return NIL;
 		}
 
-		sortedIntervalArray = SortedShardIntervalArray(finalShardIntervalList);
 		fragmentCombinationCell = list_head(fragmentCombinationList);
 
 		for (shardIndex = 0; shardIndex < finalShardCount; shardIndex++)
 		{
-			ShardInterval *shardInterval = sortedIntervalArray[shardIndex];
+			ShardInterval *shardInterval = list_nth(finalShardIntervalList, shardIndex);
```

Comment:
Note that this is an `O(n)` operation done `n` times, making this an `O(N^2)` operation. This should just be a `foreach()` loop.